src/system/game-data.ts

```diff
@@ -553,6 +570,84 @@ export class GameData {
     });
   }
 
+  public async getRunHistoryData(scene: BattleScene): Promise<Object> {
+    if (!Utils.isLocal) {
+      /**
+       * Networking Code DO NOT DELETE!
+       *
+      const response = await Utils.apiFetch("savedata/runHistory", true);
+      const data = await response.json();
+      */
+      if (localStorage.hasOwnProperty(`runHistoryData_${loggedInUser?.username}`)) {
+        let cachedResponse = localStorage.getItem(`runHistoryData_${loggedInUser?.username}`);
+        if (cachedResponse) {
+          cachedResponse = JSON.parse(decrypt(cachedResponse, true));
+        }
+        const cachedRHData = cachedResponse ?? {};
+        // check to see whether cachedData or serverData is more up-to-date
+        /**
+       * Networking Code DO NOT DELETE!
+       *
+        if ( Object.keys(cachedRHData).length >= Object.keys(data).length ) {
+          return cachedRHData;
+        }
+        */
+        return cachedRHData;
+      } else {
+        localStorage.setItem(`runHistoryData_${loggedInUser?.username}`, JSON.parse(encrypt("", true)));
+        return {};
+      }
+      //return data;
+    } else {
+      let cachedResponse = localStorage.getItem(`runHistoryData_${loggedInUser?.username}`);
+      if (cachedResponse) {
+        cachedResponse = JSON.parse(decrypt(cachedResponse, true));
+      }
+      const cachedRHData = cachedResponse ?? {};
+      return cachedRHData;
+    }
+  }
+
+  async saveRunHistory(scene: BattleScene, runEntry : SessionSaveData, victory: boolean): Promise<boolean> {
```

Comment:
```suggestion
  async saveRunHistory(scene: BattleScene, runEntry : SessionSaveData, isVictory: boolean): Promise<boolean> {
```

to make it more clear that `victory` is a boolean